src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -178,4 +202,79 @@ private String generatePassword() {
         String generatedPassword = "superSecurePassword";
         return generatedPassword;
     }
+
+    /**
+     * This function retrieves the auth token associated with a service account.
+     * Fails if the provided account is not a service account or account is not enabled.
+     *
+     * @param accountName A string representing the name of the account
+     * @return A string auth token
+     */
+    public String generateAuthToken(String accountName) throws IOException {
+
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getUserConfigName(), false);
+
+        if (!internalUsersConfiguration.exists(accountName)) {
+            throw new UserServiceException(FAILED_ACCOUNT_RETRIEVAL_MESSAGE);
+        }
+
+        String authToken = null;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode accountDetails = mapper.readTree(internalUsersConfiguration.getCEntry(accountName).toString());
```

Comment:
Can you use DefaultObjectMapper.readTree?

Reply:
Yeah will do.